### DISCUSSION TITLE
Have we over-simplified Lesson 12: DeFi Stablecoin? I don't understand the purpose of the project.

### DISCUSSION THREAD

**Discussion by alfheimrShiven** [Cyfrin/foundry-full-course-cu#879] [General]
The general benefit of minting stablecoins is to safeguard our collateral (wETH / wBTC) from market fluctuations. 

**My understanding of how stablecoin protocols work:**

```
A user deposited collateral: 1 ETH @ $2000
DSC Minting Threshold to safeguard his ETH: $2000/2 = $1000 = 1000 DSC
     * Current state: 
     * User Minted DSC = 1000 DSC, 
     * User collateral balance = 1 ETH @ $2000
```
     
!!!MARKET CRASHES!!! ETH (collateral) price drops drastically from $2000 -> $100 

     * Now since the user minted value (1000 DSC) > collateral value ($100) after the price drop, the user decides to exit the protocol, instead of depositing more collateral.
     * Redeems all minted DSC for collateral (ETH) = 1000 DSC = $1000 DSC / $100 ETH = 10 ETH
     * He gets back totally `10 ETH` @ $100/ETH, while he deposited `1ETH` @ $2000/ETH. 
     * The beauty? Even after the ETH price dropped to $100, the user ended up getting 10 ETH holding his total asset value to
     =  10 ETH * $100 = $1000.
     * If he hadn't purchased the stablecoin, his current asset value would have been 1 ETH = $100

**Project implementation:**
```

    But in the DSC protocol that we implemented in the lesson, we are just burning all the user's DSC and just returning him the collateral he deposited initially. 
    So according to the above example, all 1000 DSC will be burned and he will be able to redeem the 1 ETH of collateral he had deposited. Therefore, the final value he owns will be 1 ETH = $100 post the price drop. 
 
    The question is, then what's the point of him minting DSCs if he's gonna own the same asset of the crashed market (1 ETH = $ 100) at the end of it all. 
```
[Refer Code](https://github.com/Cyfrin/foundry-defi-stablecoin-f23/blob/main/src/DSCEngine.sol#L151)
    
   @PatrickAlphaC  Am I missing something here or is the project over-simplified purposefully?

**Comment by 0xusmanf** [comment]
I will give you my one cent on this.

> The general benefit of minting stablecoins is to safeguard our collateral (wETH / wBTC) from market fluctuations.

You want to mint a stablecoin because you want to have liquidity without selling your assets, this of course works on the assumption that asset price will go up in the future and you don't want to miss those profits for a short-term liquidity need.

> !!!MARKET CRASHES!!! ETH (collateral) price drops drastically from $2000 -> $100
> ```
>  * Now since the user minted value (1000 DSC) > collateral value ($100) after the price drop, the user decides to exit the protocol, instead of depositing more collateral.
>  * Redeems all minted DSC for collateral (ETH) = 1000 DSC = $1000 DSC / $100 ETH = 10 ETH
>  * He gets back totally `10 ETH` @ $100/ETH, while he deposited `1ETH` @ $2000/ETH. 
>  * The beauty? Even after the ETH price dropped to $100, the user ended up getting 10 ETH holding his total asset value to
>  =  10 ETH * $100 = $1000.
>  * If he hadn't purchased the stablecoin, his current asset value would have been 1 ETH = $100
>  ```

Here 1 crucial thing that needs to be considered is that the DSC is pegged to the USD based on the USD price of the collateral, if the collateral price crashes below the total value minted in DSC i.e. (!!!MARKET CRASHES!!! ETH (collateral) price drops drastically from $2000 -> $100 and total DSC minted are 1000) then the DSC will lose its pegg to USD and will no longer be 1 DSC = 1 USD.

PS: Yes the protocol is oversimplified to make it more understandable for beginners but this does not change the basic rules of how a stablecoin work.

**Reply by alfheimrShiven** [reply]
Hey @usmanfarooq91 thanks for responding. Appreciated! I guess I understand the liquidity aspect. Does that mean I can achieve liquidity through the stablecoins I mint? Could you explain this with an example?

Also, what happens when the DSC loses its peg to USD due to the collateral price drop? Do users end up losing their collateral value according to the market fluctuation? If so, then what role did stablecoin play in safeguarding the user from market volatility, as promised by stablecoins?

**Reply by 0xusmanf** [reply]
> Hey @usmanfarooq91 thanks for responding. Appreciated! I guess I understand the liquidity aspect. Does that mean I can achieve liquidity through the stablecoins I mint? Could you explain this with an example?

Let's say you have 1 ETH priced at $2000 and you need $800 to spend on something and you know you will get back $800 after a month.
So you have 2 options, 1- you sell $800 worth of ETH which is 0.4 ETH, remaining balance (0.6 ETH), or 2- you deposit 1 ETH as collateral and borrow 800 DSC = $800.
After a month 1 ETH is priced at $3000.
Case 1- You buy back $800 worth of ETH which is 0.27 ETH and you end up with 0.87 ETH worth $2610.
Case 2- You pay back your 800 DSC loan and get your 1 ETH collateral back and you end up with 1 ETH worth $3000.

> Redeems all minted DSC for collateral (ETH) = 1000 DSC = $1000 DSC / $100 ETH = 10 ETH

You don't get back your collateral based on your collateral's current price to DSC amount ratio, you are only going to get what you deposited in the first place, in this case, 1 ETH by returning borrowed DSC regardless of the current ETH price as long as your health factor is good.


> Also, what happens when the DSC loses its peg to USD due to the collateral price drop? Do users end up losing their collateral value according to the market fluctuation?

Yes, users will end up losing their collateral value in case DSC loses its peg to USD due to the collateral price crash. I say price crash because minor fluctuations and drops will not cause it to lose its peg as it is over-collateralized.

> If so, then what role did stablecoin play in safeguarding the user from market volatility, as promised by stablecoins?

Stablecoins do not safeguard you from the volatility of your underlying collateral assets, they only promise their own price stability.

**Reply by alfheimrShiven** [reply]
Thanks for the elaborative explanation @usmanfarooq91 😃.

So what I understand is that stablecoins will protect me from market fluctuation _**in case I need liquidity**._ Instead of liquidating my asset (ETH) i.e. highly volatile as compared to DSC, I _**achieve liquidity using a stablecoin,**_ which isn't volatile. 
Therefore I can return just what I borrowed, instead of returning more or less to get the same amount of asset back if I had chosen to achieve liquidity through ETH itself.
Am I understanding it right brother? 😅

**Reply by alfheimrShiven** [reply]
Q2: There is also a use case we've covered in the project, where one deposits collateral without even needing to mint DSC or achieve liquidity. Why do people do that?

**Reply by 0xusmanf** [reply]
> Thanks for the elaborative explanation @usmanfarooq91 😃.
> 
> So what I understand is that stablecoins will protect me from market fluctuation in case I need liquidity. Instead of liquidating my asset (ETH) i.e. highly volatile as compared to DSC, I achieve liquidity using a stablecoin, which isn't volatile.
> Therefore I can return just what I borrowed, instead of returning more or less to get the same amount of asset back if I had chosen to achieve liquidity through ETH itself.
> Am I understanding it right brother? 😅

You are welcome, always happy to help :). Yes, you understood it right.

> Q2: There is also a use case we've covered in the project, where one deposits collateral without even needing to mint DSC or achieve liquidity. Why do people do that?

In this specific case, there is no benefit of doing this but in real life, the user will get a certain amount of interest by depositing their ETH i.e. if you deposit 1 ETH you will get 5% interest per annum.

**Reply by alfheimrShiven** [reply]
My understanding was completely off but now I get it. Awesome. Thank you so very much @usmanfarooq91. 🙌🏾

**Reply by 0xusmanf** [reply]
Would highly recommend being part of [this](https://github.com/Cyfrin/foundry-full-course-f23/discussions/887#discussioncomment-7043942) if you want to excel in web3.
LFG 🚀 🚀 🚀 🔥  🔥  🔥

**Reply by alfheimrShiven** [reply]
Hey @usmanfarooq91 I've already filled in the form a few days back.
